pandas/core/dtypes/common.py

```diff
@@ -1448,6 +1450,9 @@ def is_extension_array_dtype(arr_or_dtype) -> bool:
     elif isinstance(dtype, np.dtype):
         return False
     else:
+        # TODO ugly -> move into registry find()? Or make this work with pandas_dtype?
+        if dtype is str and using_string_dtype():
```

Comment:
Are there other places where maybe we should just have a function like `is_builtin_string_dtype` instead of this branch? I think its hard to mentally process what logic like this is doing - would be good to abstract that as much as we can

Reply:
@WillAyd I replaced the `registry.find()` with `pandas_dtype()`, then I don't need to handle the case of `str` here specifically because `pandas_dtype()` already does that.

In general `pandas_dtype()` should be our general preprocessing function. I only needed to suppress any warning or error here, because in this case if `pandas_dtype` fails then `is_extension_array_dtype` should return `False`. 
(potentially that could be factored out into something like `pandas_dtype_safe()` if that would be needed in multiple places)